src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -401,6 +401,22 @@ private TypeGenerationSpec ParseTypeGenerationSpec(in TypeToGenerate typeToGener
                 {
                     classType = ClassType.Enum;
                 }
+                else if (IsMemoryType(type, out INamedTypeSymbol? namedMemoryType, out collectionType))
```

Comment:
I would recommend using the same approach as `byte[]`, i.e. register the two types as built-in supported:

https://github.com/dotnet/runtime/blob/d59af2cf097acb100ad6a9cba652be34be6f4a2e/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs#L387-L390

Reply:
I tried that first, but I couldn't make it work i.e: I wasn't able to register `Memory<byte>` nor `ReadOnlyMemory<byte>`.

`MemoryByteType` evaluates as `null` and I couldn't figure out why is that.
```cs
public INamedTypeSymbol? MemoryByteType => GetOrResolveType(typeof(Memory<byte>), ref _MemoryByteType);
private Option<INamedTypeSymbol?> _MemoryByteType;
```

I can follow-up on that and potentially send a separate PR to fix it.

Reply:
That's odd. Does it only happen in .NET framework by any chance?

Reply:
I noticed it by debugging `System.Text.Json.SourceGeneration.Roslyn4.4.Unit.Tests`, so I believe it happens on .NET 8 as well.

Reply:
Both 4.4 and 3.1 tests target both net80 and net462. I believe the issue is you need to add a reference to the assembly containing memory in net462:

https://github.com/dotnet/runtime/blob/00ccd49cd13961a1ade140dfbe188bc10b5049d3/src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/CompilationHelper.cs#L61

Reply:
There's already a reference to `ReadOnlySpan<>` assembly, it should be the same one. 

Note: the known-type symbol for **unbounded** `Memory<>` and `ReadOnlyMemory<>` is not null.

Reply:
Can you try passing the fully qualified name as a string directly instead of using the Type helper?

Reply:
Following this approach:

https://github.com/dotnet/runtime/blob/d59af2cf097acb100ad6a9cba652be34be6f4a2e/src/libraries/System.Text.Json/gen/Helpers/KnownTypeSymbols.cs#L135

Reply:
Actually, I have a better suggestion. You can use the [`INamedTypeSymbol.Construct`](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.inamedtypesymbol.construct?view=roslyn-dotnet-4.6.0) to apply a type argument to the generic memory types that you already have:
```
var memoryOfByte = _knownSymbols.ReadOnlyMemoryType.Construct(_compilation.GetSpecialType(SpecialType.System_Byte));
```
you can use that value in the `CreateBuiltInSupportTypeSet` method here:

https://github.com/dotnet/runtime/blob/47bb82146bb42ef881eb084160bf22e93be96f60/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs#L1471-L1490

Reply:
Nice! I've updated the PR to use this. https://github.com/dotnet/runtime/pull/88713/commits/a342a8e3c6a434f9333babe99550295ec13555ef

Reply:
Great!